Gemfile

```diff
@@ -1,2 +1,16 @@
 source "https://rubygems.org"
 gemspec
+
+group :development do
+  # This allows us to easily switch between different versions of ActiveRecord.
+  # To use this in local dev, you can do:
+  # ```
+  # rm Gemfile.lock
+  # ACTIVE_RECORD_VERSION="7.1" bundle install
+  # ```
+  active_record_version = ENV.fetch("ACTIVE_RECORD_VERSION", nil)
+  gem "activerecord", "~> #{active_record_version}.0" if active_record_version&.length&.positive?
+
+  # Just helping out the bundler resolver:
+  gem "rails", "> 6.0"
+end
```

Comment:
would it make more sense to make different groups instead of using ENV?

eg
```ruby
group :rails7_1 do
  # ...
end

group :rails7_2 do
  # ...
end
```

Reply:
The nice thing about the ENV variable is that we can set it to any version of ActiveRecord without having to update the Gemfile.  But happy to switch it to this if you prefer!

Reply:
Hmm, after running:
```
rm Gemfile.lock
bundle config set --local with activerecord_7_1
bundle install
```
I'm getting this error
```
[!] There was an error parsing `Gemfile`: You cannot specify the same gem twice with different version requirements.
You specified: activerecord (~> 7.1.0) and activerecord (~> 7.2.0). Bundler cannot continue.

 #  from /Users/aidanlavis/git/replica_pools/Gemfile:21
 #  -------------------------------------------
 #  group :activerecord_7_2, optional: true do
 >    gem "activerecord", "~> 7.2.0"
 #  end
 #  -------------------------------------------
 ```
 
 With this diff:
 ```diff
diff --git a/Gemfile b/Gemfile
index 1c249f6..0170d9d 100644
--- a/Gemfile
+++ b/Gemfile
@@ -2,15 +2,21 @@ source "https://rubygems.org"
 gemspec

 group :development do
-  # This allows us to easily switch between different versions of ActiveRecord.
-  # To use this in local dev, you can do:
-  # ```
-  # rm Gemfile.lock
-  # ACTIVE_RECORD_VERSION="7.1" bundle install
-  # ```
-  active_record_version = ENV.fetch("ACTIVE_RECORD_VERSION", nil)
-  gem "activerecord", "~> #{active_record_version}.0" if active_record_version&.length&.positive?
-
   # Just helping out the bundler resolver:
   gem "rails", "> 6.0"
 end
+
+# This allows us to easily switch between different versions of ActiveRecord.
+# To use this in local dev, you can do:
+# ```
+# rm Gemfile.lock
+# bundle config set --local with activerecord_7_1
+# bundle install
+# ```
+group :activerecord_7_1, optional: true do
+  gem "activerecord", "~> 7.1.0"
+end
+
+group :activerecord_7_2, optional: true do
+  gem "activerecord", "~> 7.2.0"
+end
```